lib/features/email/data/repository/email_repository_impl.dart

```diff
@@ -459,4 +461,39 @@ class EmailRepositoryImpl extends EmailRepository {
       emailId,
       eventActionType);
   }
+
+  @override
+  Future<List<Email>> parseEmailByBlobIds(AccountId accountId, Set<Id> blobIds) {
+    return emailDataSource[DataSourceType.network]!.parseEmailByBlobIds(accountId, blobIds);
+  }
+
+  @override
+  Future<String> generatePreviewEmailEMLContent(PreviewEmailEMLRequest previewEmailEMLRequest) {
+    return emailDataSource[DataSourceType.network]!.generatePreviewEmailEMLContent(previewEmailEMLRequest);
```

Comment:
IMO, we can reuse logic of `repository.transformEmailContent`, how about move the logic from `datasource.generatePreviewEmailEMLContent` to here?

Reply:
We should not do that, email handling in these two cases is different.